bugbug/bug_features.py

```diff
@@ -232,12 +232,18 @@ def __call__(self, bug, **kwargs):
 
 class components_touched(object):
     def __call__(self, bug, **kwargs):
-        return (ct for commit in bug['commits'] for ct in commit['components'])
+        components = set()
```

Comment:
Here and in the other function, you can create the components in one go with a list comprehension.